src/js/libgeo/GeoOps.js

```diff
@@ -584,6 +584,43 @@ geoOps.OtherPointOnCircle.updatePosition = function(el) {
     el.homog = General.withUsage(pos, "Point");
 };
 
+
+// start and end assumed to be normalized at once!
+geoOps._helper.projectPointToSegmentCR = function(seg, pos, infseg) {
+    var line = seg.homog;
+    var tt = List.turnIntoCSList([line.value[0], line.value[1], CSNumber.zero]);
+    var farpoint = List.sub(seg.startpos, seg.endpos);
+    var cr = List.crossratio3(
+        farpoint, seg.startpos, seg.endpos, pos, tt);
+
+    //handle case if we have infinite segment
+    if (infseg) {
```

Comment:
Suppose you have an arc from *A* through *B* to *C*. Then the point is uniquely identified by its cross ratio with respect to these, as seen from any other point on the circle like e.g. *I*. This means that CR(*A*,*C*,*B*,*P*) should be between 0 and ∞. If it is negative, then we should map to one of the arc endpoints. Perhaps depending on whether the cross ratio is greater or less than −1. Because if *B* plays the role of the midpoint, then its harmonic conjugate (at cross ratio −1) would play the role of the farpoint, and thus tell you which endpoint to choose. All of this should work equally well for the degenerate and the non-degenerate case, I think. Or am I missing something?

I'm slightly concerned by the fact that we have a parameter range [0,∞] here but [0,1] for PointOnLine. So this requires a homogeneous parameter vector, while that one can be encoded as a single scalar parameter. Should we make an effort to make these two more consistent somehow?